### ISSUE TITLE
/opt/rke/entrypoint.sh: no such file or directory

### ISSUE THREAD

**Issue by HighwayofLife** [rancher/rke#846]
**RKE version:**
rke version v0.1.9-rc9
rke-tools version 0.1.13, forked with a minor change to permissions that I'm testing for the cloud-provider.sh file. I've uploaded this to a private registry as tag:

```
privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
CoreOS

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Azure

**cluster.yml file:**
Kubernetes version: Default (rancher/hyperkube:v1.11.1-rancher1)

```
system_images:
  alpine: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
  nginx_proxy: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
  cert_downloader: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
  kubernetes_services_sidecar: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
```

**Steps to Reproduce:**
```
rke up --disable-port-check
```

**Results:**

```
FATA[0141] [controlPlane] Failed to bring up Control Plane:
  Failed to start [kube-apiserver] container on host [10.18.160.21]:
  Error response from daemon: invalid header field value "oci runtime error: container_linux.go:247:
  starting container process caused \"exec: \\\"/opt/rke/entrypoint.sh\\\":
  stat /opt/rke/entrypoint.sh: no such file or directory\"\n" 
```

This should have hit `/opt/rke-tools/entrypoint.sh`

**Comment by galal-hussein** [comment]
@HighwayofLife can you check if you tagged your image as v0.1.13 instead of v0.1.14 and ran rke up will cause the same error?

**Comment by moelsayed** [comment]
@HighwayofLife I retagged `0.1.13` as `0.1.14` and created a cluster on both CoreOS and ubuntu with the default k8s version. In both cases the process completed successfully. 

Can you please share the cluster file ?

**Comment by HighwayofLife** [comment]
I initially used tag `v0.1.13-1`, and after that failed, retagged and redeployed with `v0.1.14`, but I think it was still trying to use the `v0.1.13-1` version for some reason. I'm attempting to reproduce.

**Comment by HighwayofLife** [comment]
This makes sense now, initially loading in an rke-tools with a tag of `v0.1.13-1` caused the check to default to legacy entrypoint.sh, and loading in the updated version of rke-tools doesn't cause the kube-apiserver to start with a different entrypoint, so that's why it still won't work even after correcting the rke-tools tag.

**Comment by HighwayofLife** [comment]
I destroyed my dev cluster and spun up a completely fresh cluster with the following cluster.yml file, and it came back with the same (incorrect) entrypoint.sh:

```
---
nodes:
- address: 10.18.160.20
  hostname_override: devcluster-master0
  internal_address: 10.18.160.20
  role:
  - controlplane
  - etcd
  user: devadmin
- address: 10.18.160.22
  hostname_override: devcluster-master1
  internal_address: 10.18.160.22
  role:
  - controlplane
  - etcd
  user: devadmin
- address: 10.18.160.21
  hostname_override: devcluster-master2
  internal_address: 10.18.160.21
  role:
  - controlplane
  - etcd
  user: devadmin
- address: 10.18.160.19
  hostname_override: devcluster-worker0
  internal_address: 10.18.160.19
  role:
  - worker
  user: devadmin
  labels:
    app: ingress
- address: 10.18.160.17
  hostname_override: devcluster-worker1
  internal_address: 10.18.160.17
  role:
  - worker
  user: devadmin
  labels:
    app: ingress
- address: 10.18.160.18
  hostname_override: devcluster-worker2
  internal_address: 10.18.160.18
  role:
  - worker
  user: devadmin
  labels:
    app: ingress
private_registries:
- url: privateregistry.azurecr.io
  user: <redacted>
  password: <redacted>
authentication:
  strategy: x509
  sans:
  - 10.18.160.70
ssh_key_path: "../keys/<redacted>"
cloud_provider:
  name: azure
  azureCloudProvider:
    <redacted>
network:
  plugin: canal
authorization:
  mode: rbac
services:
  etcd:
  kube-api:
    service_cluster_ip_range: 10.43.0.0/16
    extra_args:
      oidc-client-id: "spn:<redacted>"
      oidc-issuer-url: "https://sts.windows.net/<redacted>"
      oidc-username-claim: "upn"
      oidc-groups-claim: "groups"
      v: 2
  kube-controller:
    cluster_cidr: 10.42.0.0/16
    service_cluster_ip_range: 10.43.0.0/16
    extra_args:
      v: 2
  scheduler:
  kubelet:
    cluster_domain: dev.<redacted>.net
    cluster_dns_server: 10.43.0.10
    infra_container_image: gcr.io/google_containers/pause-amd64:3.0
    extra_args:
      v: 2
      authentication-token-webhook: true
  kubeproxy:
ignore_docker_version: true
cluster_name: "devcluster"
ingress:
  provider: nginx
  node_selector:
    app: ingress

system_images:
  alpine: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
  nginx_proxy: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
  cert_downloader: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14
  kubernetes_services_sidecar: privateregistry.azurecr.io/rancher/rke-tools:v0.1.14

addons_include:
  - ../manifests/azure-ad-kube-rolebind.yml
```

**Comment by HighwayofLife** [comment]
```
FATA[0312] [controlPlane] Failed to bring up Control Plane:
  Failed to start [kube-apiserver] container on host [10.18.160.21]:
  Error response from daemon: OCI runtime create failed:
  container_linux.go:348: starting container process caused "exec: \"/opt/rke/entrypoint.sh\":
  stat /opt/rke/entrypoint.sh: no such file or directory": unknown 
```

```
$ docker ps -a
CONTAINER ID        IMAGE                                                COMMAND                  CREATED             STATUS                      PORTS               NAMES
211e1d33becc        rancher/hyperkube:v1.11.1-rancher1                   "/opt/rke/entrypoint…"   9 minutes ago       Created                                         kube-apiserver
d75e7fb6fa31        privateregistry.azurecr.io/rancher/rke-tools:v0.1.14   "/bin/bash"              9 minutes ago       Created                                         service-sidekick
770cc90af0ca        rancher/coreos-etcd:v3.2.18                          "/usr/local/bin/etcd…"   9 minutes ago       Up 9 minutes                                    etcd
123474293b1d        privateregistry.azurecr.io/rancher/rke-tools:v0.1.14   "/bin/bash"              13 minutes ago      Exited (0) 13 minutes ago                       cert-fetcher
```

**Comment by tomerleib** [comment]
Confirm to reproduce this issue with the following:
AWS + kubernetes 1.11 + rke-tools:v0.1.13

**Comment by tomerleib** [comment]
Here's my output of docker:

```bash
ubuntu@sandbox11:~$ docker ps -a
CONTAINER ID        IMAGE                                COMMAND                  CREATED             STATUS                     PORTS               NAMES
8bd60a85e73c        rancher/rke-tools:v0.1.13            "cert-deployer"          4 seconds ago       Exited (0) 4 seconds ago                       cert-deployer
a84170926885        rancher/rancher-agent:v2.0.6         "run.sh --server h..."   30 minutes ago      Up 30 minutes                                  loving_curie
79bf8efa0c93        rancher/rancher-agent:v2.0.6         "run.sh --server h..."   43 minutes ago      Up 30 minutes                                  clever_jones
47a8d11b0c08        rancher/hyperkube:v1.11.1-rancher1   "/opt/rke/entrypoi..."   About an hour ago   Created                                        kube-apiserver
0ff90f880a7e        rancher/rke-tools:v0.1.13            "/bin/bash"              About an hour ago   Exited (0) 5 minutes ago                       service-sidekick
7c13ca74c496        rancher/coreos-etcd:v3.2.18          "/usr/local/bin/et..."   About an hour ago   Up 30 minutes                                  etcd
ce92031d3f51        rancher/rancher-agent:v2.0.6         "run.sh --server h..."   About an hour ago   Up 30 minutes                                  romantic_mcnulty
d729d6b24aac        rancher/rancher-agent:v2.0.6         "run.sh -- share-r..."   About an hour ago   Up 30 minutes                                  share-mnt
87e9f0e6fc46        rancher/rancher-agent:v2.0.6         "run.sh --server h..."   About an hour ago   Up 30 minutes                                  youthful_carson
```
when I inspect  rancher/hyperkube:v1.11.1-rancher1  I'm seeing the reported error:
```bash
container_linux.go:247: starting container process caused "exec: \"/opt/rke/entrypoint.sh\": stat /opt/rke/entrypoint.sh: no such file or directory"
container_linux.go:247: starting container process caused "exec: \"/opt/rke/entrypoint.sh\": stat /opt/rke/entrypoint.sh: no such file or directory"
container_linux.go:247: starting container process caused "exec: \"/opt/rke/entrypoint.sh\": stat /opt/rke/entrypoint.sh: no such file or directory"
```
I've looked into a new container based on the same image and mentioned path does not exists.
Also, the rke-tools fails to start with no errors.

**Comment by galal-hussein** [comment]
@tomerleib i think you are using old rke version and that is causing the problem.

@HighwayofLife can you try with the official image `rancher/rke-tools:v0.1.14` and simply tag the image to the private registry image and try again, we can't reproduce the issue with any operating system

**Comment by tomerleib** [comment]
@galal-hussein 
rke version v0.1.8
rancher version v2.0.6

**Comment by tomerleib** [comment]
Also, here's the related part from my cluster.yml

```bash
system_images:
  etcd: rancher/coreos-etcd:v3.2.18
  alpine: rancher/rke-tools:v0.1.13
  nginx_proxy: rancher/rke-tools:v0.1.13
  cert_downloader: rancher/rke-tools:v0.1.13
  kubernetes_services_sidecar: rancher/rke-tools:v0.1.13
  kubedns: rancher/k8s-dns-kube-dns-amd64:1.14.8
  dnsmasq: rancher/k8s-dns-dnsmasq-nanny-amd64:1.14.8
  kubedns_sidecar: rancher/k8s-dns-sidecar-amd64:1.14.8
  kubedns_autoscaler: rancher/cluster-proportional-autoscaler-amd64:1.0.0
  kubernetes: rancher/hyperkube:v1.11.1-rancher1
  flannel: rancher/coreos-flannel:v0.9.1
  flannel_cni: rancher/coreos-flannel-cni:v0.2.0
  calico_node: rancher/calico-node:v3.1.1
  calico_cni: rancher/calico-cni:v3.1.1
  calico_controllers: ""
  calico_ctl: rancher/calico-ctl:v2.0.0
  canal_node: rancher/calico-node:v3.1.1
  canal_cni: rancher/calico-cni:v3.1.1
  canal_flannel: rancher/coreos-flannel:v0.9.1
  wave_node: weaveworks/weave-kube:2.1.2
  weave_cni: weaveworks/weave-npc:2.1.2
  pod_infra_container: rancher/pause-amd64:3.1
  ingress: rancher/nginx-ingress-controller:0.10.2-rancher3
  ingress_backend: rancher/nginx-ingress-controller-defaultbackend:1.4
ssh_key_path: ~/.ssh/id_rsa
ssh_agent_auth: false
authorization:
  mode: rbac
  options: {}
ignore_docker_version: false
kubernetes_version: 1.11.1
```

**Comment by galal-hussein** [comment]
@tomerleib you shouldn't use the v0.1.13 rke-tools with:
```
rke version v0.1.8
rancher version v2.0.6
```
rke-tools v0.1.13 will work only with the latest rke rc or rancher rc

**Comment by tomerleib** [comment]
@galal-hussein 
I was using the mentioned images from here: https://github.com/rancher/types/blob/master/apis/management.cattle.io/v3/k8s_defaults.go#L399

If I should use something else for 1.11.1-rancher1, I would be happy to know

**Comment by galal-hussein** [comment]
@tomerleib these defaults are for the master not for version 2.0.6, the 2.0.6 version doesn't support k8s 1.11, these are the images supported for 2.0.6:
```
https://github.com/rancher/rancher/blob/v2.0.6/vendor/github.com/rancher/types/apis/management.cattle.io/v3/k8s_defaults.go#L264
```
you can remove system images and `kubernetes_version` option and rancher will automatically pick the latest default

**Comment by tomerleib** [comment]
@galal-hussein So I can't work with K8S 1.11 as mentioned here? #https://github.com/rancher/rancher/issues/14613

**Comment by galal-hussein** [comment]
@tomerleib k8s 1.11 will be included in the next release by default (v2.0.7), you can still use the kubernetes image as a system image:
```
system_images:
  kubernetes: rancher/hyperkube:v1.11.1-rancher1
```
but you shouldn't use rke-tools v0.1.13, you should leave it blank and rke or rancher will pick up the default automatically

**Comment by tomerleib** [comment]
Removed the related items from my cluster.yml
```bash
ubuntu@ip-10-1-3-40:~$ rke up
INFO[0000] Building Kubernetes cluster                  
INFO[0000] [dialer] Setup tunnel for host [10.1.2.195]  
INFO[0000] [state] Found local kube config file, trying to get state from cluster 
INFO[0000] [reconcile] Local config is not vaild, rebuilding admin config 
INFO[0000] [reconcile] Rebuilding and updating local kube config 
INFO[0000] Successfully Deployed local admin kubeconfig at [./kube_config_cluster.yml] 
INFO[0000] [state] Fetching cluster state from Kubernetes 
INFO[0030] Timed out waiting for kubernetes cluster to get state 
INFO[0030] [network] Deploying port listener containers 
INFO[0030] [network] Pulling image [rancher/rke-tools:v0.1.10] on host [10.1.2.195] 
INFO[0035] [network] Successfully pulled image [rancher/rke-tools:v0.1.10] on host [10.1.2.195] 
INFO[0035] [network] Successfully started [rke-cp-port-listener] container on host [10.1.2.195] 
INFO[0035] [network] Successfully started [rke-worker-port-listener] container on host [10.1.2.195] 
INFO[0035] [network] Port listener containers deployed successfully 
INFO[0035] [network] Running control plane -> etcd port checks 
INFO[0036] [network] Successfully started [rke-port-checker] container on host [10.1.2.195] 
INFO[0036] [network] Running control plane -> worker port checks 
INFO[0036] [network] Successfully started [rke-port-checker] container on host [10.1.2.195] 
INFO[0036] [network] Running workers -> control plane port checks 
INFO[0036] [network] Successfully started [rke-port-checker] container on host [10.1.2.195] 
INFO[0036] [network] Checking KubeAPI port Control Plane hosts 
INFO[0036] [network] Removing port listener containers  
INFO[0036] [remove/rke-etcd-port-listener] Successfully removed container on host [10.1.2.195] 
INFO[0037] [remove/rke-cp-port-listener] Successfully removed container on host [10.1.2.195] 
INFO[0037] [remove/rke-worker-port-listener] Successfully removed container on host [10.1.2.195] 
INFO[0037] [network] Port listener containers removed successfully 
INFO[0037] [certificates] Attempting to recover certificates from backup on [etcd,controlPlane] hosts 
INFO[0037] [certificates] Successfully started [cert-fetcher] container on host [10.1.2.195] 
INFO[0042] [certificates] Certificate backup found on [etcd,controlPlane] hosts 
INFO[0042] [reconcile] Rebuilding and updating local kube config 
INFO[0042] Successfully Deployed local admin kubeconfig at [./kube_config_cluster.yml] 
INFO[0042] [reconcile] Reconciling cluster state        
INFO[0042] [reconcile] This is newly generated cluster  
INFO[0042] [certificates] Deploying kubernetes certificates to Cluster nodes 
INFO[0047] Successfully Deployed local admin kubeconfig at [./kube_config_cluster.yml] 
INFO[0047] [certificates] Successfully deployed kubernetes certificates to Cluster nodes 
INFO[0047] [cloud] Deploying cloud config file to node [10.1.2.195] 
INFO[0047] [cloud] Successfully started [cloud-config-deployer] container on host [10.1.2.195] 
INFO[0048] [remove/cloud-config-deployer] Successfully removed container on host [10.1.2.195] 
INFO[0048] [cloud] Successfully deployed kubernetes cloud config to Cluster nodes 
INFO[0048] Pre-pulling kubernetes images                
INFO[0048] Kubernetes images pulled successfully        
INFO[0048] [etcd] Building up etcd plane..              
INFO[0048] [etcd] Pulling image [rancher/coreos-etcd:v3.1.12] on host [10.1.2.195] 
INFO[0050] [etcd] Successfully pulled image [rancher/coreos-etcd:v3.1.12] on host [10.1.2.195] 
INFO[0051] [etcd] Successfully updated [etcd] container on host [10.1.2.195] 
INFO[0051] [etcd] Successfully started [rke-log-linker] container on host [10.1.2.195] 
INFO[0051] [remove/rke-log-linker] Successfully removed container on host [10.1.2.195] 
INFO[0051] [etcd] Successfully started etcd plane..     
INFO[0051] [controlplane] Building up Controller Plane.. 
INFO[0051] [sidekick] Sidekick container already created on host [10.1.2.195] 
FATA[0051] [controlPlane] Failed to bring up Control Plane: Failed to start [kube-apiserver] container on host [10.1.2.195]: Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/opt/rke/entrypoint.sh\": stat /opt/rke/entrypoint.sh: no such file or directory" 
```

**Comment by moelsayed** [comment]
@tomerleib It seems you are using the broken setup from the previous run:
```
INFO[0051] [sidekick] Sidekick container already created on host [10.1.2.195] 
```
Please use clean hosts or run `rke remove` first.

**Comment by moelsayed** [comment]
@HighwayofLife Can you please try again with `--debug` and provide `docker inspect` output the kubelet container, the sidekick container and the rke-tools image ?

**Comment by tomerleib** [comment]
@moelsayed ``` rke remove ``` followed by ```rke up`` got me passed the above error but now I've encountered this issue:
https://github.com/rancher/rke/issues/757
:no_mouth:

Let me know if I need to open a new issue on this or just leave k8s 1.11 with rancher for now...

**Comment by tomerleib** [comment]
Thanks @moelsayed , I'll start working with the rc for now.

**Comment by moelsayed** [comment]
@tomerleib Not for production! :D

**Comment by tomerleib** [comment]
Sure :+1:

**Comment by zzxwill** [comment]
Updating rke to v1.9.0 will fix the issue with kubernetes_version: v1.11.1-rancher1. 

BTW, when k8s version is v1.11.1-rancher1, updating calico-cni to v3.1.3 will also hit this issue.

**Comment by moelsayed** [comment]
@zzxwill Can you please clarify how did you hit this issue by updating calico-cni ?

**Comment by deniseschannon** [comment]
@HighwayofLife @tomerleib @zzxwill now that RKE v0.1.9 has been released and working with rke-tools:v0.1.13 with k8s v1.11.1-rancher1-1, can you let me know if you are still facing this issue. 

@zzxwill i have made a comment on the issue regarding when we upgrade to calico to make sure that we don't hit this issue when trying to upgrade it.
